node/GUIDE.md

```diff
@@ -0,0 +1,230 @@
+# Setting up a new node service with TChannel, Thrift & Hyperbahn
+
+This guide will show you how to write a server and client.
+
+The code matching this guide is [here](./examples/keyvalue/).
+
+## Dependencies
+
+Make sure you have `node` and `npm` installed. To get started
+create an empty project with `npm init` and run
+`npm install tchannel --save`.
+
+## Defining your service
+
+To define the interface of your service you can create a
+[Thrift](https://thrift.apache.org) file.
+
+Today we will use:
+
+`keyvalue.thrift`:
+```thrift
+struct GetResult {
+    1: string value
+}
+
+service KeyValue {
+    GetResult get_v1(
+        1: string key
+    )
+    void put_v1(
+        1: string key,
+        2: string value
+    )
+}
+```
+
+Our service will have two tchannel endpoints that can be.
+Namely `KeyValue::get_v1` and `KeyValue::put_v1`.
```

Comment:
Nit: Personally, I wouldn't encode the version number in the method name. There's nothing stopping you from creating multiple Thrift `service`s behind the same TChannel service since endpoints are namespaced to the service name.